plugin/trino-hive/src/main/java/io/trino/plugin/hive/PartitionsSystemTableProvider.java

```diff
@@ -68,11 +79,20 @@ public PartitionsSystemTableProvider(HivePartitionManager partitionManager)
         }
 
         SchemaTableName sourceTableName = PARTITIONS.getSourceTableName(tableName);
-        HiveTableHandle sourceTableHandle = metadata.getTableHandle(session, sourceTableName);
-
-        if (sourceTableHandle == null) {
+        Table sourceTable = metadata.getMetastore()
+                .getTable(new HiveIdentity(session), sourceTableName.getSchemaName(), sourceTableName.getTableName())
+                .orElse(null);
+        if (sourceTable == null || isDeltaLakeTable(sourceTable)) {
```

Comment:
> Disallow querying Delta Lake system tables


This is not the right commit title, since it was disallowed before.

Reply:
I thought to name it `Disallow` instead of `Deny`.
I considered that before it was denied - by throwing the exception and now it is disallowed.